src/test/scala/org/broadinstitute/dsde/workbench/sam/directory/MockDirectoryDAO.scala

```diff
@@ -214,6 +214,9 @@ class MockDirectoryDAO(private val groups: mutable.Map[WorkbenchGroupIdentity, W
     Future.successful(groupSynchronzedDates.get(groupId))
   }
 
+  override def getSynchronizedEmail(groupId: WorkbenchGroupIdentity): Future[Option[WorkbenchEmail]] = {
+    Future.successful(for (group <- groups.get(groupId)) yield group.email)
```

Comment:
One liner `for` comprehension can be changed to a `.map` https://gist.github.com/loicdescotte/4044169

Should make it slightly more readable imho